File: src/main/java/pages/BlogPage.java
package pages;

import org.junit.Assert;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

import java.util.List;


public class BlogPage extends Page {
    private List<WebElement> list;
    public BlogPage(WebDriver driver) {
        super(driver);
    }

    public void open() {
        driver.get(baseUrl);
    }

    public void assertPostWithTitleIsPublished(String title) {
        list  = driver.findElements(By.xpath("//*[contains(text(),'" + title + "')]"));
        Assert.assertTrue("Text " + title + " not found!", list.size() > 0);
    }
}

File: TestAutomation/Test/BlogPage.cs
using System;
using System.Text;
using System.Text.RegularExpressions;
using System.Threading;
using NUnit.Framework;
using OpenQA.Selenium;
using OpenQA.Selenium.Firefox;
using OpenQA.Selenium.Support.UI;
using OpenQA.Selenium.Chrome;


namespace TestAutomation.Test
{
    internal class BlogPage : Page
    {
        public BlogPage(IWebDriver driver) : base(driver)
        {
        }

        internal void assertPostAdded(object title)
        {

            //Click(By.XPath("//a[contains(.,'https://automatyzacja2016.wordpress.com/2016/09/22/nowy_wpis-ms')]"));
            //Click(By.XPath("//a[@aria-label='“nowy_wpis-ms” (Edit)']"));
            Assert.True(driver.PageSource.Contains("nowy_wpis-ms"));
            Click(By.CssSelector("img.avatar.avatar-32"));
            Click(By.CssSelector("button.ab-sign-out"));
        }
    }
}